pkg/reconciler/taskrun/taskrun.go

```diff
@@ -406,9 +406,9 @@ func (c *Reconciler) prepare(ctx context.Context, tr *v1beta1.TaskRun) (*v1beta1
 	// Propagating workspaces allows users to skip declarations
 	// In order to validate the workspace bindings we create declarations based on
 	// the workspaces provided in the task run spec. This logic is hidden behind the
-	// alpha feature gate since propagating workspaces is behind that feature gate.
+	// alpha/beta feature gate since propagating workspaces is behind the beta feature gate.
 	// In addition, we only allow this feature for embedded taskSpec.
-	if config.FromContextOrDefaults(ctx).FeatureFlags.EnableAPIFields == config.AlphaAPIFields && tr.Spec.TaskSpec != nil {
+	if config.FromContextOrDefaults(ctx).FeatureFlags.EnableAPIFields != config.StableAPIFields && tr.Spec.TaskSpec != nil {
```

Comment:
should this be `BetaAPIFields` ?

Reply:
Beta features should also run under the Alpha flag so `not stable` is a better condition? instead of `if alpha or betaAPI`?

Reply:
oh sorry I didn't notice the `!=`, that makes sense!